analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+        *     #0=stringParam
+        *          |
+        *       Block 2
+        *        #1=#0
+        *      #1 is null --+
+        *          |        |
+        *       Block 3     |
+        *       #0=Hello    |
+        *          |        |
+        *        Exit <-----+
+        */
+        const string code = """stringParam ??= "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string stringParam");
+        context.ValidateEntry(LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("stringParam"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator()
+    {
+        /*       Block 1
+         *    #0=stringParam
+         *     #0 is null
+         *        /   \
+         *       /     \
+         *   Block 2  Block 3
+         *   #1=#0  #1="Hello"
+         *       \     /
+         *       Block 4
+         *         s=#1
+         *          |
+         *        Exit
+         */
+        const string code = """string s = stringParam ?? "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string stringParam");
+        context.ValidateEntry(LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("stringParam"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullConditionalOperators()
+    {
+        /*       Block 1
+         *    #1=stringParam
+         *     #1 is null
+         *        /   \
+         *       F     T
+         *       /     \
+         *   Block 2  Block 3
+         *  #0=Length #0=stringParam (DefaultValueOperation)
+         *       \     /
+         *       Block 4
+         *      result=#1
+         *          |
+         *        Exit
+         */
+        const string code = """int? anInt =  stringParam?.Length;""";
+        var context = CreateContextCS(code, additionalParameters: "string stringParam");
+        context.ValidateEntry(LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"));
+        context.Validate(context.Cfg.Blocks[3], null); // FIXME: LiveIn("stringParam") and LiveOut("StringParam") do not appear here due to DefaultValueOperation
```

Comment:
Why is this FIXME? I would not expect `stringParam` to appear anymore. It's not read in block 4

Reply:
I forgot to remove it - it's what I asked you yesterday.